warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -2046,6 +2047,60 @@ public static final GeoTimeSerie subCycleSerie(GeoTimeSerie gts, long lastbucket
     return subCycleSerie(gts, lastbucket, buckets_per_period, overwrite, null);
   }
 
+  /**
+   * copy a slice of data from {@code srcGts} to {@code subGts}.
+   * Slice is defined by {@code scrPos} index and {@code length} to copy.
+   * The underlying arrays of subGts will be updated to new ones if length is greater than their current size.
+   * For booleans GTS, the underlying bitset is always updated.
+   */
+  private static final void copyToSubGts(GeoTimeSerie srcGts, int srcPos, GeoTimeSerie subGts, int length) {
+
+    if (null == subGts.ticks || subGts.ticks.length < length) {
+      subGts.ticks = new long[length];
+    }
+    System.arraycopy(srcGts.ticks, srcPos, subGts.ticks, 0, length);
+    // locations, if any
+    if (null != srcGts.locations) {
+      if (null == subGts.locations || subGts.locations.length < length) {
+        subGts.locations = new long[length];
+      }
+      System.arraycopy(srcGts.locations, srcPos, subGts.locations, 0, length);
+    }
+    // elevations, if any
+    if (null != srcGts.elevations) {
+      if (null == subGts.elevations || subGts.elevations.length < length) {
+        subGts.elevations = new long[length];
+      }
+      System.arraycopy(srcGts.elevations, srcPos, subGts.elevations, 0, length);
+    }
+    // value, up to value type
+    switch (srcGts.type) {
+      case LONG:
+        if (null == subGts.longValues || subGts.longValues.length < length) {
+          subGts.longValues = new long[length];
+        }
+        System.arraycopy(srcGts.longValues, srcPos, subGts.longValues, 0, length);
+        break;
+      case DOUBLE:
+        if (null == subGts.doubleValues || subGts.doubleValues.length < length) {
+          subGts.doubleValues = new double[length];
+        }
+        System.arraycopy(srcGts.doubleValues, srcPos, subGts.doubleValues, 0, length);
+        break;
+      case STRING:
+        if (null == subGts.stringValues || subGts.stringValues.length < length) {
+          subGts.stringValues = new String[length];
+        }
+        System.arraycopy(srcGts.stringValues, srcPos, subGts.stringValues, 0, length);
+        break;
+      case BOOLEAN:
+        subGts.booleanValues = srcGts.booleanValues.get(srcPos, srcPos + length);
+        break;
+    }
+    // set new length of subGts
+    subGts.values = length;
+  }
```

Comment:
I think you should also set the TYPE of subGts, or at least check that it is of the same type than srcGts and return an exception if you don't want to allow multi-type.